packages/client/hmi-client/src/utils/project-card.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * This function returns a base64 image that is dynamically generated based on the values passed into it
+ * It requires p5.js
+ */
+
+import p5 from 'p5';
+
+// Because toDataURL is not part of the official documentation
+// https://github.com/processing/p5.js/issues/2326
+interface IRenderer extends p5.Renderer {
+	toDataURL(): string;
+}
+interface Ip5 extends p5 {
+	canvas: IRenderer;
+}
+
+interface IPlaceholderArgs {
+	contributors: number;
+	models: number;
+	datasets: number;
+	papers: number;
+}
+
+// Define variables
+let placeholderCanvas: p5 | null = null;
+const imageHeight = 133;
+const imageWidth = 208;
+let contributors: number;
+let models: number;
+let datasets: number;
+let papers: number;
```

Comment:
These (contributors, models, datasets ...etc) would be better served if they are not at the global scape
- It would be easier to follow the logic, and
- it reduces the odds of creating dangling or orphaned DOM trees which might lead to memory leaks

At a glance, it seems like we just need to change a couple of function signatures to slot them in as parameters

Reply:
Sure we can do that.